src/sentry/incidents/endpoints/serializers/workflow_engine_action.py

```diff
@@ -1,19 +1,59 @@
+from collections.abc import Mapping
+from typing import Any
+
+from django.contrib.auth.models import AnonymousUser
+
 from sentry.api.serializers import Serializer
 from sentry.incidents.endpoints.serializers.alert_rule_trigger_action import (
     get_identifier_from_action,
     get_input_channel_id,
     human_desc,
 )
-from sentry.incidents.models.alert_rule import AlertRuleTriggerAction
-from sentry.notifications.models.notificationaction import ActionService
+from sentry.notifications.models.notificationaction import ActionService, ActionTarget
 from sentry.notifications.notification_action.group_type_notification_registry.handlers.metric_alert_registry_handler import (
     MetricAlertRegistryHandler,
 )
-from sentry.workflow_engine.models import Action, ActionAlertRuleTriggerAction
+from sentry.users.models.user import User
+from sentry.users.services.user.model import RpcUser
+from sentry.workflow_engine.models import (
+    Action,
+    ActionAlertRuleTriggerAction,
+    DataCondition,
+    DataConditionAlertRuleTrigger,
+    DataConditionGroupAction,
+    DetectorWorkflow,
+    WorkflowDataConditionGroup,
+)
+from sentry.workflow_engine.models.data_condition import Condition
 
 
 class WorkflowEngineActionSerializer(Serializer):
-    def serialize(self, obj: Action, attrs, user, **kwargs):
+    def get_alert_rule_trigger_id(self, action: Action) -> int:
+        """
+        Fetches the alert rule trigger id for the detector trigger related to the given action
+        """
+        action_dcga = DataConditionGroupAction.objects.get(action=action)
```

Comment:
I thought about putting try / catches around these but if we don't have this stuff that means something failed during dual write where we already have a lot of checks